Makefile

```diff
@@ -157,6 +157,12 @@ endif
 	@# Generate spec and exclude package from dependency which causes a 'classifier: unknown swagger annotation "extendee"' error.
 	@# For more details see: https://github.com/go-swagger/go-swagger/issues/2917.
 	swagger generate spec -o doc/rest-api.yaml -w ./lxd -m -x github.com/grpc-ecosystem/grpc-gateway/v2/protoc-gen-openapiv2/options
+	if [ -t 0 ] && ! git diff --quiet -- ./doc/rest-api.yaml; then \
+		read -rp "Would you like to commit swagger YAML changes (Y/n)? " answer; \
+		if [ "$${answer:-y}" = "y" ] || [ "$${answer:-y}" = "Y" ]; then \
+			git commit -sm "doc/rest-api: Refresh swagger YAML" -- ./doc/rest-api.yaml;\
```

Comment:
Do we want to also include the `-S` flag for `git commit`?

Reply:
I'm OK with adding "-S" to all of them (including the older one) but if you do provide `-S` on each commit, you'll likely want to add this to your `~/.gitconfig`:

```
# ~/.gitconfig
[commit]
    gpgsign = true
```